### PR TITLE
Resolve Skipped Tests

### PR DESCRIPTION
**Pull request by shadinaif**
This is to resolve test that were skipped in https://github.com/Edraak/edraak-platform/pull/87/commits/005252a48640dd54cdc2dc814f00de17f43ba427

One test is not resolved yet because it is related to Marketing.
`test_choose_mode_edraak_redirect_post` in `common/djangoapps/course_modes/tests/test_views.py`

### CODE REVIEW

File: common/djangoapps/student/tests/test_views.py [hunk 1]
```diff
@@ -257,7 +257,8 @@ class StudentDashboardTests(SharedModuleStoreTestCase, MilestonesTestCaseMixin,
     MOCK_SETTINGS = {
         'FEATURES': {
             'DISABLE_START_DATES': False,
-            'ENABLE_MKTG_SITE': True
+            'ENABLE_MKTG_SITE': True,
+            'EDRAAK_VIEWPORT_CHANGES': settings.FEATURES.get('EDRAAK_VIEWPORT_CHANGES', False),
```
Comment (ghost) [inline common/djangoapps/student/tests/test_views.py]: This is an edX test. Let's make the default feature `EDRAAK_VIEWPORT_CHANGES` to False, instead of patching it here.

Also, let's change is usage in the HTML templates from `FEATURES['EDRAAK_VIEWPORT_CHANGES']` to `FEATURES.get('EDRAAK_VIEWPORT_CHANGES', False)`:

```
edx-platform $ git grep EDRAAK_VIEWPORT_CHANGES
common/djangoapps/edraak_tests/tests/test_viewport_meta.py:        self.assertTrue(settings.FEATURES['EDRAAK_VIEWPORT_CHANGES'])
common/djangoapps/edraak_tests/tests/test_viewport_meta.py:    @patch.dict(settings.FEATURES, EDRAAK_VIEWPORT_CHANGES=False)
common/djangoapps/edraak_tests/tests/test_viewport_meta.py:        'EDRAAK_VIEWPORT_CHANGES',  # The feature flag should be used
lms/envs/common.py:    'EDRAAK_VIEWPORT_CHANGES': True,
lms/templates/courseware/courseware-chromeless.html:% if settings.FEATURES['EDRAAK_VIEWPORT_CHANGES']:
lms/templates/main.html:    % if not settings.FEATURES['EDRAAK_VIEWPORT_CHANGES']:
```

That should be a much better fix for the problem.

File: common/djangoapps/student/tests/test_views.py [hunk 2]
```diff
@@ -268,6 +269,7 @@ class StudentDashboardTests(SharedModuleStoreTestCase, MilestonesTestCaseMixin,
     MOCK_SETTINGS_HIDE_COURSES = {
         'FEATURES': {
             'HIDE_DASHBOARD_COURSES_UNTIL_ACTIVATED': True,
+            'EDRAAK_VIEWPORT_CHANGES': settings.FEATURES.get('EDRAAK_VIEWPORT_CHANGES', False),
```
Comment (ghost) [inline common/djangoapps/student/tests/test_views.py]: Same for :point_up:

File: openedx/core/djangoapps/user_api/tests/test_constants.py [hunk 3]
```diff
@@ -1,10 +1,31 @@
 # -*- coding: utf-8 -*-
 """Constants used in the test suite. """
 SORTED_COUNTRIES = [
+    (u"DZ", u"Algeria"),
```
Comment (ghost) [inline openedx/core/djangoapps/user_api/tests/test_constants.py]: Why this change is needed?

We have the `COUNTRIES_FIRST` setting for that:

```
$ git grep -A6 -i countries_first
lms/envs/common.py:COUNTRIES_FIRST = [
lms/envs/common.py-    'DZ',
lms/envs/common.py-    'BH',
lms/envs/common.py-    'KM',
lms/envs/common.py-    'DJ',
lms/envs/common.py-    'EG',
lms/envs/common.py-    'IQ',
--
lms/envs/common.py:COUNTRIES_FIRST_SORT = True  # Since the order depends on the name after translation
lms/envs/common.py-
lms/envs/common.py:COUNTRIES_FIRST_BREAK = '--'
lms/envs/common.py-
lms/envs/common.py-# which access.py permission name to check in order to determine if a course is visible in
lms/envs/common.py-# the course catalog. We default this to the legacy permission 'see_exists'.
lms/envs/common.py-COURSE_CATALOG_VISIBILITY_PERMISSION = 'see_exists'
lms/envs/common.py-
lms/envs/common.py-# which access.py permission name to check in order to determine if a course about page is
--
openedx/core/djangoapps/user_api/tests/test_views.py:        self.assertTrue(settings.COUNTRIES_FIRST_SORT, 'Should re-sort the COUNTRIES_FIRST for each language')
--
openedx/core/djangoapps/user_api/tests/test_views.py:                if country_code in settings.COUNTRIES_FIRST
--
openedx/core/djangoapps/user_api/tests/test_views.py:                if country_code not in settings.COUNTRIES_FIRST
```

File: openedx/core/djangoapps/user_api/tests/test_views.py [hunk 4]
```diff
@@ -1276,13 +1275,28 @@ def test_register_form_third_party_auth_running_google(
                     "value": "",
                     "default": False
                 }
+            ] + [
```
Comment (ghost) [inline openedx/core/djangoapps/user_api/tests/test_views.py]: Sounds like this is not necessity in my opinion. This is just test code that we shouldn't care about much as long as it's passing.

**Review by ghost — CHANGES_REQUESTED** [review]
@shadinaif Overall good catches. Please address the comments and ping me for another review.

File: common/djangoapps/student/tests/test_views.py (hunk 2 above)
Comment (shadinaif) [reply common/djangoapps/student/tests/test_views.py]: Done

File: openedx/core/djangoapps/user_api/tests/test_constants.py (hunk 3 above)
Comment (shadinaif) [reply openedx/core/djangoapps/user_api/tests/test_constants.py]: @OmarEdraak the problem is in `COUNTRIES_FIRST_SORT = True` . It is causing two tests to fail because the result is already sorted; and therefore, not equal to `SORTED_COUNTRIES`
`test_register_form_third_party_auth_running_google`
`test_registration_form_country`

Should we switch the default value in test environment, patch it here, or keep like this? What do you suggest?

**Review by ghost — APPROVED** [review]
Looks much better. Please turn of the sort for the countries in the tests so we can avoid the large diff. Otherwise LGTM.
